homeassistant/components/media_player/__init__.py

```diff
@@ -356,11 +358,14 @@ async def async_setup(hass: HomeAssistant, config: ConfigType) -> bool:
         SERVICE_VOLUME_SET,
         vol.All(
             cv.make_entity_service_schema(
-                {vol.Required(ATTR_MEDIA_VOLUME_LEVEL): cv.small_float}
+                {
+                    vol.Required(ATTR_MEDIA_VOLUME_LEVEL): cv.small_float,
+                    vol.Optional(ATTR_MEDIA_VOLUME_STEP): cv.small_float,
+                }
             ),
-            _rename_keys(volume=ATTR_MEDIA_VOLUME_LEVEL),
+            _rename_keys(volume=ATTR_MEDIA_VOLUME_LEVEL, step=ATTR_MEDIA_VOLUME_STEP),
         ),
-        "async_set_volume_level",
+        "async_set_volume_level_and_step",
```

Comment:
I see the challenge here. We want to set the step attribute in the base class, but if we do that in `set_volume_level`, nearly all of the platforms will need to reimplement setting the step attribute because they override this method, which is undesirable. I think introducing a function before `async_set_volume_level` makes sense, but I propose that it be a private method with the same name:

```suggestion
        "_async_set_volume_level",
```

The implementation of this method checks for the presence of `step` and sets the attribute, then calls `async_set_volume_level` as before.